src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -82,13 +80,9 @@ class AgoraBusiness {
                  entityTypes: Seq[AgoraEntityType.EntityType],
                  username: String): AgoraEntity = {
     val foundEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId)
-
-    if (!AgoraEntityPermissionsClient.getEntityPermission(foundEntity, username).canRead &&
-        !AgoraEntityPermissionsClient.getEntityPermission(foundEntity, "public").canRead) {
-      throw new AgoraEntityNotFoundException(foundEntity)
-    }
-
-    foundEntity.addUrl().removeIds()
+    val entities = AgoraEntityPermissionsClient.filterEntityByRead(Seq(foundEntity), username)
```

Comment:
this makes checking for read permissions way faster since it checks username and public in the same sql query